contrib/mesos/pkg/queue/historical.go

```diff
@@ -377,6 +377,19 @@ func (f *HistoricalFIFO) merge(id string, obj UniqueCopyable) (notifications []E
 	return
 }
 
+// Resync will touch all objects to put them into the processing queue
+func (f *HistoricalFIFO) Resync() error {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+	for id, entry := range f.items {
+		if entry.Is(ADD_EVENT|UPDATE_EVENT) {
+			f.queue = append(f.queue, id)
+		}
+	}
+	f.cond.Broadcast()
+	return nil
```

Comment:
Sorry I think I'm missing the whole point of this Resync func. Can someone give me additional context? This implementation is more correct in the sense that it's putting the correct types into internal data structures now. But ADD and UPDATE events should already be queued so this feels somewhat redundant. What am I missing?